lib/membrane_webm/parser.ex

```diff
@@ -0,0 +1,276 @@
+defmodule Membrane.WebM.Parser do
+  use Membrane.Filter
+
+  alias Membrane.{Buffer, Time}
+  alias Membrane.WebM.Parser.Vint
+  alias Membrane.WebM.Schema
+
+  def_input_pad :input,
+    availability: :always,
+    mode: :pull,
+    demand_unit: :buffers,
+    caps: :any
+
+  def_output_pad :output,
+    availability: :always,
+    mode: :pull,
+    caps: :any
+
+  @return_elements [
+    :EBML,
+    :SeekHead,
+    :Info,
+    :Tracks,
+    :Tags,
+    :Cues,
+    :Cluster
+  ]
+
+  @impl true
+  def handle_init(_) do
+    {:ok, %{acc: <<>>}}
+  end
+
+  @impl true
+  def handle_demand(:output, size, :buffers, _context, state) do
+    {{:ok, demand: {:input, size}}, state}
+  end
+
+  @impl true
+  def handle_process(:input, %Buffer{payload: payload}, _context, %{acc: acc}) do
+    unparsed = acc <> payload
+
+    case parse_many(unparsed, []) do
+      {:ok, {name, _data} = parsed_element, rest} ->
+        IO.puts("Parser sending #{name}")
+        {{:ok, buffer: {:output, %Buffer{payload: parsed_element}}}, %{acc: rest}}
```

Comment:
As mentioned previously, `data` probably should be the payload and `name` should be put in metadata.